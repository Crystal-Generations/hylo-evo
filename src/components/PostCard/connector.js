import { connect } from 'react-redux'
import { fetchPost } from 'store/actions/posts'
import { getPostById } from 'store/selectors/getPostById'

export function mapStateToProps (state, props) {
  return {
    post: getPostById(state, props)
  }
}

export const mapDispatchToProps = { fetchPost }

export default connect(mapStateToProps, mapDispatchToProps)
