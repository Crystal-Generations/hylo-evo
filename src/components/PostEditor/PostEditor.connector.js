import { connect } from 'react-redux'
import getParam from 'store/selectors/getParam'
import getMe from 'store/selectors/getMe'
import getPost from 'store/selectors/getPost'
import getCommunityForCurrentRoute from 'store/selectors/getCommunityForCurrentRoute'
import {
  createPost,
  updatePost
} from './PostEditor.store'

export function mapStateToProps (state, props) {
  const currentUser = getMe(state)
  const communityOptions = props.communityOptions || (currentUser &&
    currentUser.memberships.toModelArray().map(m => m.community))
  const editing = !!getParam('postId', state, props)
  let post = props.post || getPost(state, props)
  const loading = editing && !post
  const currentCommunity = getCommunityForCurrentRoute(state, props)
  if (!editing && currentCommunity) {
    post = {communities: [currentCommunity]}
  }
  return {
    post,
    communityOptions,
    currentUser,
    editing,
    loading
  }
}

export const mapDispatchToProps = (dispatch, props) => {
  return {
    updatePost: (postParams) => dispatch(updatePost(postParams)),
    createPost: (postParams) => dispatch(createPost(postParams))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
