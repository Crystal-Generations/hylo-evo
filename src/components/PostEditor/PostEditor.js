import React, { PropTypes } from 'react'
import cx from 'classnames'
import styles from './PostEditor.scss'
import Avatar from 'components/Avatar'
import HyloEditor from 'components/HyloEditor'
import Button from 'components/Button'
import CommunitiesSelector from 'components/CommunitiesSelector'

const DEFAULT_TITLE_PLACEHOLDER = 'Whatâ€™s on your mind?'
const DEFAULT_BODY_PLACEHOLDER = 'Add a description'

export default class PostEditor extends React.Component {
  static propTypes = {
    titlePlaceholder: PropTypes.string,
    bodyPlaceholder: PropTypes.string
  }

  static defaultProps = {
    titlePlaceholder: DEFAULT_TITLE_PLACEHOLDER,
    bodyPlaceholder: DEFAULT_BODY_PLACEHOLDER
  }

  constructor (props) {
    super(props)
    this.state = {
      title: '',
      titlePlaceholder: props.titlePlaceholder,
      postType: 'discussion'
    }
  }

  handleTitleChange = (event) => this.setState({title: event.target.value})

  handlePostTypeSelection = postType => event => {
    let titlePlaceholder
    switch (postType) {
      case 'discussions':
        titlePlaceholder = DEFAULT_TITLE_PLACEHOLDER
        break
      case 'request':
        titlePlaceholder = DEFAULT_TITLE_PLACEHOLDER
        break
      case 'offer':
        titlePlaceholder = 'What super powers can you offer?'
        break
      default:
        titlePlaceholder = DEFAULT_TITLE_PLACEHOLDER
    }
    this.setState({titlePlaceholder, postType})
  }

  postTypeButtonProps = type => {
    const { postType } = this.state
    return {
      label: type,
      onClick: this.handlePostTypeSelection(type),
      className: cx(
        styles.postType,
        styles[`postType-${type}`],
        {
          [styles[`postType-${type}-active`]]: postType === type
        }
      )
    }
  }

  setSelectedCommunities = selectedCommunities => this.setState({ selectedCommunities })

  save = () => {
    const { postType, selectedCommunities } = this.state
    const results = {
      postType,
      selectedCommunities
    }
    console.log(results)
    return results
  }

  render () {
    const { bodyPlaceholder } = this.props
    const { titlePlaceholder, title } = this.state

    return <div styleName='wrapper'>
      <div styleName='body'>
        <div styleName='initialPrompt'>What are you looking to post?</div>
        <div styleName='postTypes'>
          <Button {...this.postTypeButtonProps('discussion')} />
          <Button {...this.postTypeButtonProps('request')} />
          <Button {...this.postTypeButtonProps('offer')} />
        </div>
        <div styleName='titleAndDescription'>
          <div>
            <Avatar
              medium
              styleName='titleAvatar'
              url=''
              avatarUrl='https://d3ngex8q79bk55.cloudfront.net/user/13986/avatar/1444260480878_AxolotlPic.png' />
          </div>
          <div>
            <input
              type='text'
              styleName='title-input'
              placeholder={titlePlaceholder}
              value={title}
              onChange={this.handleTitleChange} />
            <HyloEditor styleName='editor' placeholder={bodyPlaceholder} />
          </div>
        </div>
      </div>
      <div styleName='footer'>
        <div styleName='postIn'>
          <div styleName='postIn-label'>Post in</div>
          <div styleName='postIn-communities'>
            <CommunitiesSelector onChange={this.setSelectedCommunities} />
          </div>
        </div>
        <div styleName='actionsBar'>
          <div styleName='actions' />
          <Button onClick={this.save} styleName='postButton' label='Post' color='green' />
        </div>
      </div>
    </div>
  }
}
