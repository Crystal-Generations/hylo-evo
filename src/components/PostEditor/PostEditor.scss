// mixins

.section-label {
  color: $color-rhino-80;
  white-space: nowrap;
  margin-right: 28px;
  font-size: 14px;
}

// Overall wrapper

.wrapper {
  display: flex;
  flex-direction: column;
  min-width: $post-editor-width;
  list-style: none;
  border: solid 1px rgb(0,0,0);
  border-radius: $space-1x;
  border: 1px solid $color-cape-cod-10;
  background: $color-white;
}

// Header area

.header {
  padding: $space-4x $space-4x 0 $space-4x;
}

.initial {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  line-height: normal;
  margin-bottom: $space-4x;
  &-prompt {
    composes: bdy-lt-sm from 'css/typography.scss';
  }
  &-closeButton {
    cursor: pointer;
  }
}

.postTypes {
  display: flex;
  flex-wrap: nowrap;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: $space-6x;
}

.postType {
  display: inline-block;
  width: 100%;
  margin: 0 $space-2x;
  padding: $space-5x;
  font-size: 0.625rem;
  text-transform: uppercase;
  text-align: center;
  line-height: 0;
  border-radius: $space-1x;
  color: $color-rhino-30;
  background-color: $color-athens-gray-dark;
  &:last-child {
    margin-right: 0;
  }
  &:first-child {
    margin-left: 0;
  }
  &-discussion {
    &.selectable:hover, &:active, &.active {
      color: $color-picton-blue;
      background-color: $color-link-water;
    }
  }
  &-request {
    &.selectable:hover, &:active, &.active {
      color: $color-persimmon;
      background-color: $color-peach-schnapps;
    }
  }
  &-offer {
    &.selectable:hover, &:active, &.active  {
      color: $color-caribbean-green;
      background-color: $color-iceberg;
    }
  }
}

// Body area

.body {
  padding: 0 $space-4x;
  display: flex;
  flex-direction: row;
  &-column:last-child {
    width: 100%;
  }
}

.titleAvatar {
  margin-right: $space-2x;
  padding: $space-1x;
  height: $space-9x;
}

.titleInput {
  composes: hdr-headline from 'css/typography.scss';
  margin-bottom: $space-2x;
  padding: 0;
  height: $space-9x;
  width: 100%;
  outline: none;
  border: none;
  &:disabled {
    color: $color-rhino-80;
  }
  &::placeholder {
    color: $color-rhino-30;
  }
}

.editor {
  height: 200px;
  margin-bottom: $space-6x;
  color: $color-rhino-80;
}

.linkPreview {
  height: $space-16x;
  background-color: $color-medium-purple;
}

// Footer area

.footer {
  padding: 0;
}

.postIn {
  min-height: 3rem;
  border-top: solid 1px $color-ghost-30;
  padding: $space-5x;
  line-height: 1;
  display: flex;
  align-items: center;
  &-label {
    composes: section-label;
  }
  &-communities {
    /* !empty */
  }
}

.actionsBar {
  display: flex;
  flex-wrap: nowrap;
  flex-direction: row;
  padding: $space-3x 18px;
  align-items: center;
  border-top: solid 1px $color-ghost-30;
}

.actions {

}

.action-icon {
  font-size: 32px;
  cursor: pointer;
  &:hover {
    color: $color-caribbean-green;
  }
}

.highlight-icon {
  color: $color-caribbean-green;
  cursor: default;
}

.postButton {
  margin-left: auto;
}

// ImagePreviews

.image-preview-section {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 15px 20px 0px;
  border-top: solid 1px $color-ghost-30;
}

.image-previews {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
}

.image-box {
  width: 100px;
  height: 100px;
  border-radius: 4px;
  margin-right: 15px;
  margin-bottom: 15px;
}

.image-preview {
  composes: image-box;
}

.image {
  composes: image-box;
  background-size: cover;
  background-position: center;
  position: relative;
  padding: 8px;
}

.remove-button {
  position: absolute;
  top: -7px;
  right: -7px;
  border-radius: 50%;
  color: white;
  background-color: $color-rhino;
  font-size: 12px;
  padding: 1px;
  border: 1px solid white;
  background-clip: padding-box;
  cursor: pointer;
}

.drag-preview {
  width: 84px;
  height: 84px;
}

.add-image {
  composes: image-box;
  padding-top: 16px;
  text-align: center;
  border: 2px dashed $color-loblolly;
  font-size: 40px;
  color: $color-rhino-30;
  cursor: pointer;
}
