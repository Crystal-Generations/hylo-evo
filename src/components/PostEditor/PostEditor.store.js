import { get, pullAt, clone } from 'lodash/fp'
import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'
import linkMatcher from 'util/linkMatcher'

export const MODULE_NAME = 'PostEditor'
export const CREATE_POST = `${MODULE_NAME}/CREATE_POST`
export const UPDATE_POST = `${MODULE_NAME}/UPDATE_POST`
export const UPDATE_POST_PENDING = `${UPDATE_POST}_PENDING`
export const FETCH_LINK_PREVIEW = `${MODULE_NAME}/FETCH_LINK_PREVIEW`
export const REMOVE_LINK_PREVIEW = `${MODULE_NAME}/REMOVE_LINK_PREVIEW`
export const CLEAR_LINK_PREVIEW = `${MODULE_NAME}/CLEAR_LINK_PREVIEW`
export const LOAD_IMAGE_PREVIEWS = `${MODULE_NAME}/LOAD_IMAGE_PREVIEWS`
export const SET_IMAGE_PREVIEWS = `${MODULE_NAME}/SET_IMAGE_PREVIEWS`
export const ADD_IMAGE_PREVIEW = `${MODULE_NAME}/ADD_IMAGE_PREVIEW`
export const REMOVE_IMAGE_PREVIEW = `${MODULE_NAME}/REMOVE_IMAGE_PREVIEW`
export const SWITCH_IMAGE_PREVIEWS = `${MODULE_NAME}/SWITCH_IMAGE_PREVIEWS`

// Actions

export function createPost (post) {
  const { type, title, details, communities, linkPreview, imageUrls } = post
  const linkPreviewId = linkPreview && linkPreview.id
  const communityIds = communities.map(c => c.id)
  return {
    type: CREATE_POST,
    graphql: {
      query: `mutation ($type: String, $title: String, $details: String, $linkPreviewId: String, $communityIds: [String], $imageUrls: [String]) {
        createPost(data: {type: $type, title: $title, details: $details, linkPreviewId: $linkPreviewId, communityIds: $communityIds, imageUrls: $imageUrls}) {
          id
          type
          title
          details
          commentersTotal
          communities {
            id
            name
            slug
          }
          creator {
            id
          }
          linkPreview {
            id
          }
          attachments {
            id
            type
            url
            position
          }
        }
      }`,
      variables: {
        type,
        title,
        details,
        linkPreviewId,
        communityIds,
        imageUrls
      }
    },
    meta: {extractModel: 'Post'}
  }
}

export function updatePost (post) {
  const { id, type, title, details, communities, linkPreview, imageUrls } = post
  const linkPreviewId = linkPreview && linkPreview.id
  const communityIds = communities.map(c => c.id)
  return {
    type: UPDATE_POST,
    graphql: {
      query: `mutation ($id: ID, $type: String, $title: String, $details: String, $linkPreviewId: String, $communityIds: [String], $imageUrls: [String]) {
        updatePost(id: $id, data: {type: $type, title: $title, details: $details, linkPreviewId: $linkPreviewId, communityIds: $communityIds, imageUrls: $imageUrls}) {
          id
          type
          title
          details
          linkPreview {
            id
          }
          communities {
            id
            name
            slug
          }
          attachments {
            id
            type
            url
            position
          }
        }
      }`,
      variables: {
        id,
        type,
        title,
        details,
        linkPreviewId,
        communityIds,
        imageUrls
      }
    },
    meta: {
      id,
      extractModel: {
        modelName: 'Post',
        getRoot: get('updatePost'),
        append: false
      }
    }
  }
}

export function fetchLinkPreview (url) {
  return {
    type: FETCH_LINK_PREVIEW,
    graphql: {
      query: `mutation ($url: String) {
        findOrCreateLinkPreviewByUrl(data: {url: $url}) {
          id
          url
          imageUrl
          title
          description
          imageWidth
          imageHeight
          status
        }
      }`,
      variables: {
        url
      }
    },
    meta: {
      extractModel: {
        modelName: 'LinkPreview',
        getRoot: get('findOrCreateLinkPreviewByUrl')
      }
    }
  }
}

export function pollingFetchLinkPreview (dispatch, htmlContent) {
  const poll = (url, delay) => {
    if (delay > 4) return
    dispatch(fetchLinkPreview(url)).then(value => {
      if (!value) return
      const linkPreviewFound = value.meta.extractModel.getRoot(value.payload.data)
      if (!linkPreviewFound) {
        setTimeout(() => poll(url, delay * 2), delay * 1000)
      }
    })
  }
  if (linkMatcher.test(htmlContent)) {
    const urlMatch = linkMatcher.match(htmlContent)[0].url
    poll(urlMatch, 0.5)
  }
}

export function removeLinkPreview () {
  return {type: REMOVE_LINK_PREVIEW}
}

export function clearLinkPreview () {
  return {type: CLEAR_LINK_PREVIEW}
}

export function setImagePreviews (imagePreviews) {
  return {
    type: SET_IMAGE_PREVIEWS,
    payload: imagePreviews
  }
}

export function addImagePreview (url) {
  return {
    type: ADD_IMAGE_PREVIEW,
    payload: url
  }
}

export function removeImagePreview (position) {
  return {
    type: REMOVE_IMAGE_PREVIEW,
    payload: position
  }
}

export function switchImagePreviews (position1, position2) {
  return {
    type: SWITCH_IMAGE_PREVIEWS,
    payload: {
      position1,
      position2
    }
  }
}

// Selectors

export const getLinkPreview = ormCreateSelector(
  orm,
  state => state.orm,
  state => state[MODULE_NAME],
  ({ LinkPreview }, { linkPreviewId }) =>
    LinkPreview.hasId(linkPreviewId) ? LinkPreview.withId(linkPreviewId).ref : null
)

export function getImagePreviews (state) {
  return state[MODULE_NAME].imagePreviews
}

export const getImages = ormCreateSelector(
  orm,
  get('orm'),
  (state, { postId }) => postId,
  ({ Attachment }, postId) =>
    Attachment.all()
    .filter(({ type, post }) =>
      type === 'image' && post === postId)
    .orderBy('position')
    .toModelArray()
  )

// Reducer

export const defaultState = {
  linkPreviewId: null,
  linkPreviewStatus: null,
  imagePreviews: []
}

export default function reducer (state = defaultState, action) {
  const { error, type, payload, meta } = action
  if (error) return state

  switch (type) {
    case FETCH_LINK_PREVIEW:
      const linkPreview = (meta.extractModel.getRoot(payload.data))
      if (linkPreview && !linkPreview.title) {
        return {...state, linkPreviewId: null, linkPreviewStatus: 'invalid'}
      }
      return {...state, linkPreviewId: get('id')(linkPreview), linkPreviewStatus: null}
    case REMOVE_LINK_PREVIEW:
      return {...state, linkPreviewId: null, linkPreviewStatus: 'removed'}
    case CLEAR_LINK_PREVIEW:
      return {...state, linkPreviewId: null, linkPreviewStatus: 'cleared'}
    case SET_IMAGE_PREVIEWS:
      return {...state, imagePreviews: payload}
    case ADD_IMAGE_PREVIEW:
      return {...state, imagePreviews: state.imagePreviews.concat([payload])}
    case REMOVE_IMAGE_PREVIEW:
      return {...state, imagePreviews: pullAt(payload, state.imagePreviews)}
    case SWITCH_IMAGE_PREVIEWS:
      const { position1, position2 } = payload
      const tmp = state.imagePreviews[position1]
      const imagePreviews = clone(state.imagePreviews)
      imagePreviews[position1] = imagePreviews[position2]
      imagePreviews[position2] = tmp
      return {
        ...state,
        imagePreviews
      }
    default:
      return state
  }
}
