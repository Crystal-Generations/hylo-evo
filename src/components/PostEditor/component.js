import React, { PropTypes } from 'react'
import cx from 'classnames'
import styles from './component.scss'
import Avatar from 'components/Avatar'
import HyloEditor from 'components/HyloEditor'
import Button from 'components/Button'
import CommunitiesSelector from 'components/CommunitiesSelector'

const TITLE_PLACEHOLDER = 'Whatâ€™s on your mind?'
const BODY_PLACEHOLDER = 'Add a description'

export default class PostEditor extends React.Component {
  static propTypes = {
    titlePlaceholder: PropTypes.string,
    bodyPlaceholder: PropTypes.string
  }

  static defaultProps = {
    titlePlaceholder: TITLE_PLACEHOLDER,
    bodyPlaceholder: BODY_PLACEHOLDER
  }

  constructor (props) {
    super(props)
    this.state = {
      title: '',
      postType: 'discussion'
    }
  }

  handleTitleChange = (event) => this.setState({title: event.target.value})

  handlePostTypeSelection = type => event => this.setState({postType: type})

  render () {
    const { titlePlaceholder, bodyPlaceholder } = this.props
    const { title, postType } = this.state

    const postTypeButtonProps = type => ({
      label: type,
      onClick: this.handlePostTypeSelection(type),
      className: cx(
        styles.postType,
        styles[`postType-${type}`],
        {
          [styles[`postType-${type}-active`]]: postType === type
        }
      )
    })

    return <div styleName='wrapper'>
      <div styleName='body'>
        <div styleName='initialPrompt' className='bdy-lt-sm'>What are you looking to post?</div>
        <div styleName='postTypes'>
          <Button {...postTypeButtonProps('discussion')} />
          <Button {...postTypeButtonProps('request')} />
          <Button {...postTypeButtonProps('offer')} />
        </div>
        <div styleName='title'>
          <Avatar medium
            styleName='title-avatar'
            url=''
            avatarUrl='https://d3ngex8q79bk55.cloudfront.net/user/13986/avatar/1444260480878_AxolotlPic.png' />
          <input type='text'
            styleName='title-input'
            placeholder={titlePlaceholder}
            value={title}
            onChange={this.handleTitleChange} />
        </div>
        <HyloEditor styleName='editor' debug placeholder={bodyPlaceholder} />
      </div>
      <div styleName='footer'>
        <div styleName='postIn'>
          <div styleName='postIn-label'>
            Post in
          </div>
          <div styleName='postIn-communities'>
            <CommunitiesSelector />
          </div>
        </div>
        <div styleName='actionsBar'>
          <div styleName='actions' />
          <Button styleName='postButton' label='Post' color='green' />
        </div>
      </div>
    </div>
  }
}
