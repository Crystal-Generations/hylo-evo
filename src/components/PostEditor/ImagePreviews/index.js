import component from './ImagePreviews'
import connector from './ImagePreviews.connector'

export default connector(component)
