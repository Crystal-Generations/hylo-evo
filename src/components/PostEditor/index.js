import { HOLOCHAIN_ACTIVE } from 'util/holochain'
import component from './PostEditor'
import holochainConnector from './PostEditor.holochain.connector'
import connector from './PostEditor.connector'

export default HOLOCHAIN_ACTIVE
  ? holochainConnector(component)
  : connector(component)
