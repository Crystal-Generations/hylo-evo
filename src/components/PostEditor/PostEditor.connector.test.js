import { mapStateToProps, mapDispatchToProps, mergeProps } from './PostEditor.connector'
import orm from 'store/models'
import {
  CREATE_POST,
  MODULE_NAME,
  FETCH_LINK_PREVIEW,
  createPost,
  updatePost,
  pollingFetchLinkPreview,
  removeLinkPreview,
  clearLinkPreview,
  getLinkPreview
} from './PostEditor.store'

let state
beforeAll(() => {
  const session = orm.session(orm.getEmptyState())
  const community = session.Community.create({id: '99', slug: 'foo'})
  session.LinkPreview.create({
    id: 1
  })

  session.Me.create({
    id: '1',
    memberships: [session.Membership.create({
      id: '345',
      community: community.id,
      hasModeratorRole: true
    })]
  })

  state = {
    orm: session.state,
    PostEditor: {
      linkPreviewStatus: false
    },
    pending: {
      FETCH_LINK_PREVIEW: false
    },
    AttachmentManager: {
      image: [],
      file: []
    }
  }
})

describe('mapStateToProps', () => {
  it('returns the right keys for a new post', () => {
    const props = {
      forNew: true,
      match: {
        params: {
          slug: 'foo'
        }
      }
    }
    expect(mapStateToProps(state, props)).toMatchSnapshot()
  })
})
