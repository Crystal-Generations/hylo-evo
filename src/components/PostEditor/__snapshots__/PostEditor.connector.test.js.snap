// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapStateToProps returns the right keys 1`] = `
Object {
  "communityOptions": Array [
    SessionBoundModel {
      "_fields": Object {
        "id": "99",
        "members": undefined,
        "moderators": undefined,
        "posts": undefined,
        "slug": "foo",
      },
      "slug": "foo",
    },
  ],
  "currentCommunity": SessionBoundModel {
    "_fields": Object {
      "id": "99",
      "members": undefined,
      "moderators": undefined,
      "posts": undefined,
      "slug": "foo",
    },
    "slug": "foo",
  },
  "currentUser": SessionBoundModel {
    "_fields": Object {
      "id": "1",
      "messageThreads": undefined,
      "notifications": undefined,
      "posts": undefined,
      "skills": undefined,
    },
    "id": "1",
  },
  "editing": false,
  "fetchLinkPreviewPending": undefined,
  "files": undefined,
  "images": undefined,
  "linkPreview": null,
  "linkPreviewStatus": undefined,
  "loading": false,
  "post": null,
  "postPending": false,
  "showFiles": false,
  "showImages": false,
}
`;
